### PR TITLE
Adding product types to order creation

### LINKED ISSUE #9783: Update order creation event properties to track extensions
We want to track the usage of extensions through order creation. This task involves adding the `product_types` property to the order creation event. 

event: `ORDER_CREATE_BUTTON_TAPPED`
property `product_types` -> a string property containing all product types inside an order divided by ','
Example: `"product_types": "bundle, variable"`

### PR DESCRIPTION
**Pull request by atorresveiga**
Closes: #9783

### Why
We want to see how many orders are created in the app with product extensions after we start supporting them.

### Description
This PR adds the `PRODUCT_TYPES` property to the `ORDER_CREATE_BUTTON_TAPPED` event so we can track all the product types contained in the order created

### Testing instructions
TC1
1. Open the orders, tap
2. Add a new order
3. Add several products
4. Tap on create order
5. Check that the `ORDER_CREATE_BUTTON_TAPPED` event is triggered with the `PRODUCT_TYPES` property

TC2
1. Open the orders, tap
2. Add a new order
3. Add a fee
4. Tap on create order
5. Check that the `ORDER_CREATE_BUTTON_TAPPED` event is triggered WITHOUT the `PRODUCT_TYPES` property

### Images/gif
```
🔵 Tracked: order_create_button_tapped, Properties: {"status":null,"product_types":"variable","product_count":1,"has_customer_details":false,"has_fees":false,"has_shipping_method":false,"blog_id":214253715,"is_wpcom_store":true,"was_ecommerce_trial":false,"plan_product_slug":"ecommerce-bundle","is_debug":true,"site_url":"https:\/\/superlativecentaur.wpcomstaging.com"}
```

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditViewModel.kt [hunk 1]
```diff
@@ -929,16 +931,22 @@ class OrderCreateEditViewModel @Inject constructor(
...
-                KEY_HAS_SHIPPING_METHOD to _orderDraft.value.shippingLines.isNotEmpty()
+        launch {
+            val ids = products.value?.map { orderProduct -> orderProduct.item.productId }
+            val productTypes = if (!ids.isNullOrEmpty()) orderDetailRepository.getUniqueProductTypes(ids) else null
+            tracker.track(
+                ORDER_CREATE_BUTTON_TAPPED,
+                buildMap {
+                    put(KEY_STATUS, _orderDraft.value.status)
+                    putIfNotNull(PRODUCT_TYPES to productTypes)
+                    put(KEY_PRODUCT_COUNT, products.value?.count())
```
Comment (ThomazFB) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditViewModel.kt]: Should we also use `putIfNotNull` here too?

**Review by ThomazFB — APPROVED** [review]
Looks good!

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditViewModel.kt (hunk 1 above)
Comment (atorresveiga) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditViewModel.kt]: Good catch! Fixed here: https://github.com/woocommerce/woocommerce-android/pull/9784/commits/2743ecede9a1f151ebbc1c048b394f099f6a78f6

Comment (atorresveiga) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditViewModel.kt]: Second though, Thomaz. I believe that in the case of product count, it makes more sense to track `0` for orders not containing products. WDYT? 

Fixed here: https://github.com/woocommerce/woocommerce-android/pull/9784/commits/9205244ef4ddfc22d927534b484b857da41a5e58

Comment (ThomazFB) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/OrderCreateEditViewModel.kt]: Nice idea, @atorresveiga! Thanks for submitting my suggestion!

**Pull request merged**